web/docs/language/features.md

```diff
@@ -178,138 +178,249 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
 
 ## Queries and Actions (aka Operations)
 
-In Wasp, main interaction between client and server happens via Operations, of which two types exist: Queries and Actions.
+In Wasp, the client and the server interact with each other through Operations.
+Wasp currently supports two kinds of operations: **Queries** and **Actions**.
 
 ### Query
 
-Queries are NodeJS functions that don't modify any state. Normally they fetch certain resources, process them and return result. They are executed on server. 
+Queries are resource-fetching functions that don't modify the state. Each Query should fetch, process, and return a particular resource.
 
-To create a Wasp Query, we need two things: declaration in Wasp and implementation in NodeJS:
+Queries are implemented in NodeJS and executed within the server's context.
+However, Wasp will generate the code that lets you call the Query from any point in your code (i.e., client, server) using the same interface.
+In other words, you won't have to worry about building an HTTP API for the Query, handling the request on the server, or even handling and caching the responses on the client.
+Instead, simply focus on the business logic inside your Query and let Wasp take care of the rest!
 
-1. `query` declaration in Wasp:
-```c title="main.wasp"
-// ...
-query getTasks {
-  fn: import { getAllTasks } from "@ext/foo.js"
+To create a Wasp Query, you must:
+1. Define the Query's NodeJS implementation
+2. Declare the Query in Wasp using the `query` declaration
+
+After completing these two steps, you'll be able to use the Query from any point in your code. 
+
+
+#### Defining the Query's NodeJS implementation
+A Query must be implemented as an `async` NodeJS function that takes two arguments.
+Since both arguments are positional, you can name the parameters however you want, but we'll stick with `args` and `context`:
+1. `args`:  An object containing all the arguments (i.e., payload) sent to the Query (e.g., filtering conditions).
+Take a look at [the examples of usage](#using-the-query) to see how to pass this object to the Query.
+3. `context`: An additional context object. This object contains user session information, as well as information about entities. The examples here won't use the context for simplicity purposes. You can read more about it in the [section about using entities in queries](#using-entities-in-queries).
+
+Here's an example of two simple Queries:
+```js title="ext/queries.js"
+// our "database"
+const tasks = [
+  { id: 1, description: "Buy some eggs", isDone: true },
+  { id: 2, description: "Make an omelette", isDone: false },
+  { id: 3, description: "Eat breakfast", isDone: false }
+]
+
+
+// You don't need to use the arguments if you don't need them
+export const getAllTasks = async () => {
+  return tasks;
+}
+
+// The 'args' object is something sent by the caller (most often from the client) 
```

Comment:
Aha nice I see now you mentioned it here that args are coming from the caller! I would still maybe mention it up there, at least a bit.